include/deal.II/physics/elasticity/standard_tensors.h

```diff
@@ -0,0 +1,360 @@
+#define dealii__elasticity__standard_tensors_h
+
+
+#include <deal.II/base/symmetric_tensor.h>
+#include <deal.II/base/tensor.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Physics
+{
+
+  namespace Elasticity
+  {
+
+    /**
+     * A collection of tensor definitions that conform to notation used in
+     * standard scientific literature. These hold specifically for the
+     * codimension 0 case, where the metric tensor is the identity tensor.
+     *
+     * @relates Tensor
+     * @relates SymmetricTensor
+     * @author Jean-Paul Pelteret, Andrew McBride, 2016
+    */
+    template <int dim>
+    class StandardTensors
+    {
+    public:
+
+      /**
+       * @name Metric tensors
+       */
+//@{
+
+      /**
+       * The second-order referential/spatial symmetric identity (metric) tensor
+       * $\mathbf{I}$.
+       *
+       * This is defined such that, for any second-order Tensor or SymmetricTensor,
+       * the following holds:
+       * @f[
+       *  \mathbf{I} \cdot \{ \bullet \} = \{ \bullet \} \cdot \mathbf{I} =
+       * \{ \bullet \}
+       *    \qquad \text{and} \qquad
+       * \mathbf{I} : \{ \bullet \} = \textrm{trace} \{ \bullet \} \, .
+       * @f]
+       * This definition aligns with the second-order SymmetricTensor returned by
+       * unit_symmetric_tensor().
+       */
+      static const SymmetricTensor<2, dim> I;
+
+      /**
+       * The fourth order referential/spatial unit SymmetricTensor $\mathcal{I}$.
+       *
+       * This is defined such that, for any second-order SymmetricTensor,
+       * the following holds:
+       * @f[
+       *  \mathcal{I} : \{ \bullet \} = \{ \bullet \} : \mathcal{I} = \{ \bullet \}
+       * @f]
+       * Furthermore, for a general Tensor
+       * @f[
+          \mathcal{I} : \{ \hat{\bullet} \} =
+          \dfrac{1}{2}[\{ \hat{\bullet} \} + \{ \hat{\bullet} \}^T] \, .
+       * @f]
+       * This definition aligns with the fourth-order SymmetricTensor $\mathcal{S}$
+       * mentioned in the class description and that returned by identity_tensor().
+       *
+       * @note Since the symmetry of the operand is assumed, this fourth-order unit
+       * tensor should only be used to operate on symmetric second-order tensors.
+       */
+      static const SymmetricTensor<4, dim> II;
+
+      /**
+       * The fourth-order referential/spatial tensor $\mathbf{I} \otimes \mathbf{I}$.
+       *
+       * This is defined such that, for any second-order SymmetricTensor,
+       * the following holds:
+       * @f[
+       *  [\mathbf{I} \otimes \mathbf{I}] : \{ \bullet \} =
+       *  \textrm{trace}\{ \bullet \} \mathbf{I} \, .
+       * @f]
+       */
+      static const SymmetricTensor<4, dim> IxI;
+
+//@}
+
+      /**
+       * @name Projection operators
+       */
+//@{
+
+      /**
+       * The fourth-order spatial deviatoric tensor. Also known as the deviatoric
+       * operator, this tensor projects a second-order SymmetricTensor onto a
```

Comment:
I technically state this in the definition of the unit SymmetricTensor `II` that I provide in this namespace. But I've also now added some additional documentation to the definition of the projection tensor itself. Hopefully nobody will be left scratching their heads about this point!